DatabaseServer/database_server.py

```diff
@@ -341,22 +350,21 @@ def _get_iocs_not_to_stop() -> list:
 
     # Initialise IOC database connection
     try:
-        ioc_data = IOCData(IocDataSource(SQLAbstraction("iocdb", "iocdb", "$iocdb")), ProcServWrapper(),
-                           MACROS["$(MYPVPREFIX)"])
+        ioc_data = init_ioc_database_connection()
         print_and_log("Connected to IOCData database", INFO_MSG, LOG_TARGET)
-    except Exception as e:
+    except MaxAttemptsExceededException as ex:
         ioc_data = None
-        print_and_log("Problem initialising IOCData DB connection: {}".format(traceback.format_exc()),
-                      MAJOR_MSG, LOG_TARGET)
+        exception_info = "".join(traceback.format_exception(type(ex.args[0]), ex.args[0], ex.args[0].__traceback__))
```

Comment:
Accessing `ex.args[]` in this way is a bit horrible, it's really an implementation detail of python. I think we should override `MaxAttemptsExceededException` to have an _explicit_ `.cause` attribute, and maybe a `format_cause_traceback` method that returns an appropriate message (as a string). 

This would also then let us remove the duplication of this complicated line which is currently repeated below.